File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcDistributionElementType/IfcDistributionFlowElementType/IfcFlowControllerType/IfcElectricTimeControlType/IfcElectricTimeControlType.vb
Public Class IfcElectricTimeControlType Inherits IfcFlowControllerType
    Public PredefinedType As IfcElectricTimeControlTypeEnum

    ' === WHERE CLAUSES ===
    ' CorrectPredefinedType : (PredefinedType <> IfcElectricTimeControlTypeEnum.USERDEFINED) OR ((PredefinedType = IfcElectricTimeControlTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
End Class

File: src/main/resources/output/IFC4X3_java/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProduct/IfcElementType/IfcDistributionElementType/IfcDistributionFlowElementType/IfcFlowControllerType/IfcElectricTimeControlType/IfcElectricTimeControlType.java
public class IfcElectricTimeControlType extends IfcFlowControllerType {
    public IfcElectricTimeControlTypeEnum PredefinedType;

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : (PredefinedType <> IfcElectricTimeControlTypeEnum.USERDEFINED) OR ((PredefinedType = IfcElectricTimeControlTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcElementType.ElementType))
}
